### DISCUSSION TITLE
How to play video on the github page

### DISCUSSION THREAD

**Discussion by Mr-MRF-Dev** [community/community#23522] [Other Feature Feedback, Questions, & Ideas]
<p>Hi guys. I have a one minute video that I want to show in github page… My HTML file works well on my computer but it does not work on github page…</p>
<p>what should I do?</p>

**Comment by jfullstackdev** [comment]
<p>I cannot see it at your public repos</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/2/f/2fafcf71b915d4304f5f13788402d5b3cd13a887.png" data-download-href="/uploads/short-url/6NR70nYQNRSSx3lzTpDe4b8FFBl.png?dl=1" title="image"><img src="https://user-images.githubusercontent.com/47092464/181077515-68deefef-6230-4fbf-91c3-910bb1bf78e3.png" alt="image" data-base62-sha1="6NR70nYQNRSSx3lzTpDe4b8FFBl" width="690" height="373" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/2/f/2fafcf71b915d4304f5f13788402d5b3cd13a887_2_690x373.png, https://aws1.discourse-cdn.com/github/optimized/3X/2/f/2fafcf71b915d4304f5f13788402d5b3cd13a887_2_1035x559.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/3X/2/f/2fafcf71b915d4304f5f13788402d5b3cd13a887_2_1380x746.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/2/f/2fafcf71b915d4304f5f13788402d5b3cd13a887_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use href="#far-image"></use></svg><span class="filename">image</span><span class="informations">1920×1040 156 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use href="#discourse-expand"></use></svg>
</div></a></div>
<p>if you can make a test repo for that, so we can see it, or more details about that problem</p>
<p>usually it’s URL problem</p>

**Comment by Mr-MRF-Dev** [comment]
<p>I have a test project.</p>
<blockquote>
<p><a href="http://Mr-MRF-Dev.github.io" rel="noopener nofollow ugc">Mr-MRF-Dev.github.io</a><br>
Mr-MRF-Dev/Mr-MRF-Dev.github.io</p>
</blockquote>
<p>I uploaded a one-minute movie in the box on the right (the movie file is in the “asset” folder). When I run this project on my computer, the movie is played automatically (according to the code) but there is only a movie poster on the github.</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/7/2/7274a728e815f49481808be6f6bd38e1876a1e5b.jpeg" data-download-href="/uploads/short-url/gkwmhLWhAMN6N7h8OJA6HVS2Gw3.jpeg?dl=1" title="Screenshot-1" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/79112136/181077518-167ce682-35c8-44eb-87f2-653bb9441b7e.jpeg" alt="Screenshot-1" data-base62-sha1="gkwmhLWhAMN6N7h8OJA6HVS2Gw3" width="690" height="322" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/7/2/7274a728e815f49481808be6f6bd38e1876a1e5b_2_690x322.jpeg, https://aws1.discourse-cdn.com/github/optimized/3X/7/2/7274a728e815f49481808be6f6bd38e1876a1e5b_2_1035x483.jpeg 1.5x, https://aws1.discourse-cdn.com/github/original/3X/7/2/7274a728e815f49481808be6f6bd38e1876a1e5b.jpeg 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/7/2/7274a728e815f49481808be6f6bd38e1876a1e5b_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use href="#far-image"></use></svg><span class="filename">Screenshot-1</span><span class="informations">1354×632 72.8 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use href="#discourse-expand"></use></svg>
</div></a></div><br>
<div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/1/5/151b51b194f5e78502690e18f475c24008611626.jpeg" data-download-href="/uploads/short-url/30IyAf1rN5ZAB6CQzIg2jryhnwi.jpeg?dl=1" title="Screenshot-2" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/79112136/181077520-722a97d2-5c5a-4599-bb62-6312722066ef.jpeg" alt="Screenshot-2" data-base62-sha1="30IyAf1rN5ZAB6CQzIg2jryhnwi" width="690" height="312" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/1/5/151b51b194f5e78502690e18f475c24008611626_2_690x312.jpeg, https://aws1.discourse-cdn.com/github/original/3X/1/5/151b51b194f5e78502690e18f475c24008611626.jpeg 1.5x, https://aws1.discourse-cdn.com/github/original/3X/1/5/151b51b194f5e78502690e18f475c24008611626.jpeg 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/1/5/151b51b194f5e78502690e18f475c24008611626_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use href="#far-image"></use></svg><span class="filename">Screenshot-2</span><span class="informations">966×438 63.3 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by jfullstackdev** [comment]
<p>I access directly both files of the video, they are both playing, so we are sure, it’s accessible</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/d/e/deb3233315379ed4684a1b161c6efc05cd20c8e6.jpeg" data-download-href="/uploads/short-url/vM5NwqLo0qUy98bdCZElkV45Mfc.jpeg?dl=1" title="image"><img src="https://user-images.githubusercontent.com/47092464/181077522-2f656625-309c-498c-9d1e-2b9ab926c4c3.jpeg" alt="image" data-base62-sha1="vM5NwqLo0qUy98bdCZElkV45Mfc" width="690" height="472" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/d/e/deb3233315379ed4684a1b161c6efc05cd20c8e6_2_690x472.jpeg, https://aws1.discourse-cdn.com/github/optimized/3X/d/e/deb3233315379ed4684a1b161c6efc05cd20c8e6_2_1035x708.jpeg 1.5x, https://aws1.discourse-cdn.com/github/original/3X/d/e/deb3233315379ed4684a1b161c6efc05cd20c8e6.jpeg 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/d/e/deb3233315379ed4684a1b161c6efc05cd20c8e6_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use href="#far-image"></use></svg><span class="filename">image</span><span class="informations">1045×715 44.3 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use href="#discourse-expand"></use></svg>
</div></a></div>
<p>no errors at the Developers Tool,</p>
<p><img src="https://user-images.githubusercontent.com/47092464/181077523-95f4a1c0-36e6-4bee-b4b5-e3f050eaa52b.png" alt="image" data-base62-sha1="83TKHxF8xsTmpm7qDn5OrJiTBD1" width="486" height="367"></p>
<p>maybe I need to fork that, to test it, we’ll see</p>

**Comment by jfullstackdev** [comment]
<p>finally,</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/5/4/54beea3024c127604ab13c59a1911d26664785b3.jpeg" data-download-href="/uploads/short-url/c5H8Ip7O8ffsl1KS5NjqAkKfwu7.jpeg?dl=1" title="image"><img src="https://user-images.githubusercontent.com/47092464/181077526-67479445-86a4-4207-9e6f-d5d3ca10d55c.jpeg" alt="image" data-base62-sha1="c5H8Ip7O8ffsl1KS5NjqAkKfwu7" width="690" height="373" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/5/4/54beea3024c127604ab13c59a1911d26664785b3_2_690x373.jpeg, https://aws1.discourse-cdn.com/github/optimized/3X/5/4/54beea3024c127604ab13c59a1911d26664785b3_2_1035x559.jpeg 1.5x, https://aws1.discourse-cdn.com/github/optimized/3X/5/4/54beea3024c127604ab13c59a1911d26664785b3_2_1380x746.jpeg 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/5/4/54beea3024c127604ab13c59a1911d26664785b3_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use href="#far-image"></use></svg><span class="filename">image</span><span class="informations">1920×1040 145 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use href="#discourse-expand"></use></svg>
</div></a></div>
<p>it’s not just auto-play on GitHub and there is the warning at the Developer Tools, if you did not reload it, the warning will not show</p>
<pre><code class="lang-auto">Error with Permissions-Policy header: Unrecognized feature: 'interest-cohort'.
</code></pre>
<p>and you may want to check this out</p>
<aside class="quote" data-post="1" data-topic="179780">
  <div class="title">
    <div class="quote-controls"></div>
    
    <a href="https://github.community/t/i-have-no-idea-what-the-interest-cohort-is/179780">I have no idea what the interest-cohort is</a> <a class="badge-wrapper  bullet" href="/c/github-help/github-pages/31"><span class="badge-category-parent-bg" style="background-color: #6a737d;"></span><span class="badge-category-bg" style="background-color: #c8e1ff;"></span><span style="" data-drop-close="true" class="badge-category clear-badge" title="Questions and discussion about creating a website directly from a repository on GitHub.com.">GitHub Pages</span></a>
  </div>
  <blockquote>
    why am I getting this error “Error with Permissions-Policy header: Unrecognized feature: ‘interest-cohort’.”
  </blockquote>
</aside>

<p>but if you right click the video and click <code>Show all controls</code>, it will now play</p>
<p><img src="https://user-images.githubusercontent.com/47092464/181077527-f0961668-152f-4019-9651-4e38cf5e0892.jpeg" alt="image" data-base62-sha1="ly1VzP5Z1kgDfw1cwBrRyfPsA3S" width="497" height="420"></p>
<p><img src="https://user-images.githubusercontent.com/47092464/181077528-99e69b4a-3e43-4de2-bf90-e1badcad5c5b.jpeg" alt="image" data-base62-sha1="9aCn0aFM3OUHUxbndNEvwIKWJbN" width="400" height="377"></p>

**Comment by Mr-MRF-Dev** [comment]
<p>Thanks for the help :star_struck:</p>
<p>What do you think I should do to ensure that it does not have control and is played automatically?</p>

**Comment by jfullstackdev** [comment]
<aside class="quote no-group" data-username="Mr-MRF-Dev" data-post="6" data-topic="248995">
<div class="title">
<div class="quote-controls"></div>
 Mr-MRF-Dev:</div>
<blockquote>
<p>that it does not have control and is played automatically?</p>
</blockquote>
</aside>
<p>there are suggestions at the thread, you can try them</p>

**Comment by jfullstackdev** [comment]
<p>using Edge, there is already the controls</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/3/c/3c3700ac78a8c836c3f2efb4d357fe96c0ccbd1c.jpeg" data-download-href="/uploads/short-url/8AGuqIB4tcKcdXvuyVq0JS8w8B6.jpeg?dl=1" title="image"><img src="https://user-images.githubusercontent.com/47092464/181077533-3f13ac9c-4bf0-4fac-b44f-7f5fdaaa9dc7.jpeg" alt="image" data-base62-sha1="8AGuqIB4tcKcdXvuyVq0JS8w8B6" width="690" height="373" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/3/c/3c3700ac78a8c836c3f2efb4d357fe96c0ccbd1c_2_690x373.jpeg, https://aws1.discourse-cdn.com/github/optimized/3X/3/c/3c3700ac78a8c836c3f2efb4d357fe96c0ccbd1c_2_1035x559.jpeg 1.5x, https://aws1.discourse-cdn.com/github/optimized/3X/3/c/3c3700ac78a8c836c3f2efb4d357fe96c0ccbd1c_2_1380x746.jpeg 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/3/c/3c3700ac78a8c836c3f2efb4d357fe96c0ccbd1c_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use href="#far-image"></use></svg><span class="filename">image</span><span class="informations">1920×1040 125 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use href="#discourse-expand"></use></svg>
</div></a></div>
<p>unfortunately, it’s still not auto-play,</p>

**Comment by jfullstackdev** [comment]
<p>I tried it locally, it’s not auto-play, even without being hosted on GitHub Pages,</p>
<p>then I separate another local test just for the auto-play, still it’s not auto-play and even the examples at w3schools,</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/a/c/ac63a3d4cbf4baaa22d5110ba9a6cd6be8923422.png" data-download-href="/uploads/short-url/oB1BmeLqGysb6hbgOE3MA8XACwW.png?dl=1" title="image"><img src="https://user-images.githubusercontent.com/47092464/181077534-7d978064-0444-4ce9-8266-dd4019137e91.png" alt="image" data-base62-sha1="oB1BmeLqGysb6hbgOE3MA8XACwW" width="690" height="418" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/a/c/ac63a3d4cbf4baaa22d5110ba9a6cd6be8923422_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use href="#far-image"></use></svg><span class="filename">image</span><span class="informations">696×422 16.1 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by Mr-MRF-Dev** [comment]
<p>I used this tag from site W3, it works in the site it self, it may not work sometimes, I do not know why it does not work, but the syntax is correct (in my opinion)</p>
<blockquote>
<p><a href="https://www.w3schools.com/tags/att_video_autoplay.asp" class="inline-onebox" rel="noopener nofollow ugc">HTML video autoplay Attribute</a></p>
</blockquote>

**Comment by jfullstackdev** [comment]
<p>yup, that’s exactly what I tried</p>
<p>now, at that sample it will run, but if you try it even at local, then it will not</p>

**Comment by Mr-MRF-Dev** [comment, marked as answer]
<p>Part of the description of the W3 site: (Read the <strong>Note</strong>)</p>
<blockquote>
<h2>Definition and Usage</h2>
<p>The <code>autoplay</code> attribute is a boolean attribute.</p>
<p>When present, the video will automatically start playing.</p>
<blockquote>
<p><strong>Note:</strong> Chromium browsers do not allow autoplay in most cases. However, muted autoplay is always allowed.</p>
</blockquote>
<p>Add <code>muted</code> after <code>autoplay</code> to let your video file start playing automatically (but muted).</p>
</blockquote>

**Comment by jfullstackdev** [comment]
<p>yup, I tried it out too, mute it,</p>
<p><code>controls autoplay muted</code></p>
<p><img src="https://user-images.githubusercontent.com/47092464/181077538-3fc28403-4ae1-46b9-8d3b-4537f62215bf.png" alt="image" data-base62-sha1="bOcNiLYrLwGHJcGoLSUeQ1WebNC" width="635" height="390"></p>
<p>unfortunately, the sound is important too :rofl:</p>

**Comment by Mr-MRF-Dev** [comment]
<p>The sound of this video is more important than the video itself. :confused: :smiling_face_with_tear:</p>
<p>Anyway, thank you for helping me :star_struck:</p>

**Comment by jfullstackdev** [comment]
<p>I’m finding out why, and it might be this article</p>
<aside class="onebox allowlistedgeneric" data-onebox-src="https://developer.chrome.com/blog/autoplay/">
  <header class="source">
      <img src="https://user-images.githubusercontent.com/47092464/181077540-8664bc05-07db-400d-b436-834b680349b1.png" class="site-icon" width="32" height="32">

      <a href="https://developer.chrome.com/blog/autoplay/" target="_blank" rel="noopener">Chrome Developers</a>
  </header>

  <article class="onebox-body">
    <img src="https://user-images.githubusercontent.com/47092464/181077542-18ca1e1e-f2c1-49cd-84fa-ae6d5ec0bca1.png" class="thumbnail onebox-avatar" width="499" height="499" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/b/d/bd91c6fd3b2611acb12771a7a134a14c70d4a87c_2_499x499.png, https://aws1.discourse-cdn.com/github/optimized/3X/b/d/bd91c6fd3b2611acb12771a7a134a14c70d4a87c_2_748x748.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/3X/b/d/bd91c6fd3b2611acb12771a7a134a14c70d4a87c_2_998x998.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/b/d/bd91c6fd3b2611acb12771a7a134a14c70d4a87c_2_10x10.png">

<h3><a href="https://developer.chrome.com/blog/autoplay/" target="_blank" rel="noopener">Autoplay policy in Chrome - Chrome Developers</a></h3>

  <p>Learn best practices for good user experiences with the new autoplay policies in Chrome.</p>


  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>

<p>but yes, I’m turning off autoplay even in Twitter :rofl:</p>

**Reply by kp-444** [reply]
Hello, thank you for raising this issue. I am facing a similar issue wherein the video is not able to play on github.io while it works fine in my computer.
I tried autoplay and mute with loop but it is not working. Any suggestions?

**Comment by josephbrown98** [comment]
I’ve run into similar issues before, and a quick fix for me was uploading the video to a cloud storage service and linking it from there. It is desirable to compress the video beforehand (I do it in Movavi). You can also do it locally, but I think it's more complicated.
